packages/web-pkg/src/components/SideBar/types.ts

```diff
@@ -1,13 +1,28 @@
 import { defineComponent } from 'vue'
 import { IconFillType } from 'design-system/src/helpers'
+import { Item } from '@ownclouders/web-client/src/helpers'
 
-export interface Panel {
-  app: string
+export interface SideBarPanelContext<P, T extends Item> {
+  parent?: P
+  items?: T[]
+}
```

Comment:
Just some unfiltered thought process:

Do we need the space in the future? I'm a bit torn. On one hand, it's quite files-specific since we don't have a space in the admin-settings e.g. But then again, you could say the same about `parent`.

The only reason to add it here would be to have it available in `isVisible` and `isRoot`, right? Might be helpful if I want my panel to be visible in project spaces only. It could also save us from some of the not-so-nice route checks.

Reply:
Trying to keep the generic scope, I added a `root` field to the context interface and set the space as `root` wherever possible. Not using it, yet, but I agree that it makes a whole lot of sense to have the space root available.